internal/systemd/systemd.go

```diff
@@ -48,3 +50,37 @@ func WaitOnDevices(devs []string, stage string) error {
 
 	return nil
 }
+
+// GetSystemdVersion fetches the version of Systemd
+// in a given system.
+func GetSystemdVersion() (uint, error) {
+	var systemdVersion string
+	var begin, end int
+	conn, err := dbus.NewSystemdConnection()
+	if err != nil {
+		return 0, err
+	}
+	version, err := conn.GetManagerProperty("Version")
+	if err != nil {
+		return 0, err
+	}
+	// Handle different systemd versioning schemes that are being returned.
+	// for e.g:
+	// - Fedora 31: `"v243.5-1.fc31"`
+	// - RHEL 8: `"239"`
+	if strings.HasPrefix(version, "v") {
+		begin = 1
+	} else {
+		begin = 0
+	}
+	end = strings.Index(version, ".")
+	if end == -1 {
+		end = len(version)
+	}
+	systemdVersion = version[begin+2 : end]
```

Comment:
why was this switched to `+2` ?

Reply:
Because of the below issue:
```
Ignition failed: failed to create units: strconv.Atoi: parsing "v243": invalid syntax
```
I came across this issue when I was trying to boot-up an `FCOS` system with an ignition binary build using the changes where `systemdVersion = version[begin : end]`.

Reply:
We can make use of `regex` to capture the exact o/p. wdyt?